pkg/storage/local.go

```diff
@@ -684,6 +668,28 @@ func (is *ImageStoreLocal) PutImageManifest(repo, reference, mediaType string,
 		return "", err
 	}
 
+	// apply linter only on images, not signatures
+	if is.linter != nil {
+		if mediaType == ispec.MediaTypeImageManifest &&
+			// check that image manifest is not cosign signature
+			!strings.HasPrefix(reference, "sha256-") &&
+			!strings.HasSuffix(reference, remote.SignatureTagSuffix) {
+			// lint new index with new manifest before writing to disk
+			is.Unlock(&lockLatency)
+			pass, err := is.linter.Lint(repo, mDigest, is)
+			is.Lock(&lockLatency)
+			if err != nil {
+				is.log.Error().Err(err).Msg("linter error")
```

Comment:
you can cover this by using a mocked linter which returns an error

see mockedImageStore, mockedRoutesHandler